src/signals_notebook/users/group.py

```diff
@@ -0,0 +1,225 @@
+            )
+
+            result = GroupResponse(**response.json())  # type: ignore
+            yield from [cast(ResponseData, item).body for item in result.data]
+
+    @classmethod
+    def get(cls, group_id: str) -> 'Group':
+        """Get group by id
+
+        Args:
+            group_id: Unique user group identifier
+        Returns:
+            Group
+        """
+        api = SignalsNotebookApi.get_default_api()
+        response = api.call(
+            method='GET',
+            path=(cls._get_endpoint(), group_id),
+        )
+        result = GroupResponse(**response.json())
+        return cast(ResponseData, result.data).body
+
+    @classmethod
+    def create(cls, request: GroupRequestBody) -> 'Group':
+        """Create new Group
+
+        Returns:
+            Group
+        """
+        api = SignalsNotebookApi.get_default_api()
+        log.debug('Create User: %s...', cls.__name__)
+
+        response = api.call(
+            method='POST',
+            path=(cls._get_endpoint(),),
+            json={
+                'data': {
+                    'attributes': request.dict(by_alias=True),
+                }
+            },
+        )
+        log.debug('Group: %s was created.', cls.__name__)
+
+        result = GroupResponse(**response.json())
+        return cast(ResponseData, result.data).body
+
+    def save(self, force: bool = True) -> None:
+        """Update attributes of a specified group.
+
+        Returns:
+
+        """
+        api = SignalsNotebookApi.get_default_api()
+        api.call(
+            method='PATCH',
+            path=(
+                self._get_endpoint(),
+                self.id,
+            ),
+            params={
+                'force': json.dumps(force),
+            },
+            json={
+                'data': {
+                    'attributes': self.dict(
+                        include={
+                            'name',
+                            'description',
+                        },
+                    )
+                },
+            },
+        )
+
+    def delete(self):
+        """Delete user group by id.
+
+        Returns:
+
+        """
+        api = SignalsNotebookApi.get_default_api()
+        log.debug('Disable Group: %s...', self.id)
+
+        api.call(
+            method='DELETE',
+            path=(self._get_endpoint(), self.id),
+        )
+        log.debug('Group: %s was disabled successfully', self.id)
+
+    def get_members(self) -> list[GroupMember]:
+        """Get user group members
+
+        Returns:
+            Group members
+        """
+        api = SignalsNotebookApi.get_default_api()
+        response = api.call(
+            method='GET',
+            path=(self._get_endpoint(), self.id, 'members'),
+        )
+        result = GroupMemberResponse(**response.json())
+        return [cast(ResponseData, item).body for item in result.data]
+
+    def add_user(self, user: User, force: bool = True) -> list[GroupMember]:
```

Comment:
In my opinion if we make _member and _member_by_id attributes, we will be able to use code logic for patching members like in tables/ samples/ stoichiometry entities.